modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -517,24 +495,23 @@ private void addXmlHintsModelProperty(ExtensionParameter extensionParameter, Par
         }
     }
 
-    private void checkAnnotationsNotUsedMoreThanOnce(List<ExtensionParameter> parameters, ParameterizedDeclarer component, Class<? extends Annotation>... annotations)
+    private void checkAnnotationsNotUsedMoreThanOnce(List<ExtensionParameter> parameters, Class<? extends Annotation>... annotations)
     {
-        stream(annotations).forEach(
-                annotation ->
-                {
-                    final long count = parameters.stream().filter(param -> param.isAnnotatedWith(annotation)).count();
-                    if (count > 1)
-                    {
-                        throw new IllegalModelDefinitionException(format("The defined parameter from [%s], uses the annotation @%s more than once",
-                                                                         parameters.get(0).getOwner(),
-                                                                         annotation.getName()));
-                    }
-                });
+        for (Class<? extends Annotation> annotation : annotations)
+        {
+            final long count = parameters.stream().filter(param -> param.isAnnotatedWith(annotation)).count();
+            if (count > 1)
+            {
+                throw new IllegalModelDefinitionException(format("The defined parameter from [%s], uses the annotation @%s more than once",
```

Comment:
The defined parameter from -> Parameter '%s'. Also include the parameter's owner, the message is not useful if I don't know where that parameter is defined